app/code/Magento/InventoryConfiguration/Model/SaveStockItemConfiguration.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryConfiguration\Model;
+
+use Magento\CatalogInventory\Api\Data\StockItemInterface;
+use Magento\Framework\App\ResourceConnection;
+use Magento\InventoryCatalog\Model\GetProductIdsBySkusInterface;
+use Magento\InventoryConfigurationApi\Api\Data\StockItemConfigurationInterface;
+use Magento\InventoryConfigurationApi\Api\SaveStockItemConfigurationInterface;
+use Magento\Framework\Exception\LocalizedException;
+
+/**
+ * @inheritdoc
+ */
+class SaveStockItemConfiguration implements SaveStockItemConfigurationInterface
+{
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @var GetProductIdsBySkusInterface
+     */
+    private $getProductIdsBySkus;
+
+    public function __construct(
+        ResourceConnection $resourceConnection,
+        GetProductIdsBySkusInterface $getProductIdsBySkus
+    ) {
+        $this->resourceConnection = $resourceConnection;
+        $this->getProductIdsBySkus = $getProductIdsBySkus;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute(string $sku, StockItemConfigurationInterface $stockItemConfiguration)
```

Comment:
Our Get Service for Stock Item configuration looks like 
```php
interface GetStockItemConfigurationInterface
{
    /**
     * Return null if configuration for sku per stock is not exist
     *
     * @param string $sku
     * @param int $stockId
     * @return StockItemConfigurationInterface|null
     * @throws \Magento\Framework\Exception\LocalizedException
     */
    public function execute(string $sku, int $stockId);
}
```
So, we define the scope with $sku and $stockId

existing Save service doesn't have $stockId, so all the config data would be saved for Default stock.

```
           ->addColumn(
                'stock_id',
                \Magento\Framework\DB\Ddl\Table::TYPE_SMALLINT,
                null,
                ['unsigned' => true, 'nullable' => false, 'default' => '0'],
                'Stock Id'
            )
```
in `cataloginventory_stock_item`

Also, we need to remove unneeded constants 
```
interface StockItemConfigurationInterface
{
    const SKU = 'sku';
    const STOCK_ID = 'stock_id';
```